### PR TITLE
Upgraded to clap v4

### LINKED ISSUE #515: Upgrade clap to 4.0
While we have upgraded clap to v4 in some of our crates, it's still on v3 in `shuttle-deployer` and `cargo-shuttle`. v4 of clap has been out for a while now and it would be nice to get the benefits of upgrading, as well as having a consistent version of clap across the workspace (we can also inherit the dependency from the root manifest when these upgrades are done). 

Here is the changelog for clap v4, it also includes a guide for migrating from v3 to v4: https://github.com/clap-rs/clap/blob/master/CHANGELOG.md#400---2022-09-28

It would be preferable for this to be done in separate PRs (deployer should be the easiest). For information on how to build and test the crates locally, please refer to our [contribution guide](https://github.com/shuttle-hq/shuttle/blob/main/CONTRIBUTING.md). And of course, if anything is unclear or you have questions, don't hesitate to reach out in our discord!

- [ ] Upgrade to clap v4 in deployer
- [ ] Upgrade to clap v4 in cargo-shuttle 
- [ ] Upgrade to clap v4 in provisioner

### PR DESCRIPTION
**Pull request by gautamprikshit1**
Fixed #515

### CODE REVIEW

File: cargo-shuttle/src/args.rs [hunk 1]
```diff
@@ -38,12 +39,7 @@ pub struct Args {
 #[derive(Parser, Debug)]
 pub struct ProjectArgs {
     /// Specify the working directory
-    #[clap(
-        global = true,
-        long,
-        parse(try_from_os_str = parse_path),
-        default_value = ".",
-    )]
+    #[clap(global = true, long, value_parser, default_value = ".")]
```
Comment (chesedo) [inline cargo-shuttle/src/args.rs:42]: Shouldn't the `clap` also become `arg`?

**Review by chesedo — COMMENTED** [review]
Great, thanks for this @gautamprikshit1! Just a quick question :smile:

**Review by oddgrd — COMMENTED** [review]
Thanks for getting started on this! To add to what @chesedo said, if you follow these steps from nr 4: https://github.com/clap-rs/clap/blob/master/CHANGELOG.md#migrating you should get deprecation warnings.

**Review by oddgrd — COMMENTED** [review]
This looks great @gautamprikshit1 , you work fast! :partying_face: There are a few things we need to fix, but nothing major. Some of these changes will compile, but won't work when running the commands, so it's a good idea to run the commands and see that everything works as expected. You can do this with `cargo run --bin cargo-shuttle -- <command>`.

**Comment by gautamprikshit1** [comment]
Thanks, for the suggestions I will do this asap

**Review by oddgrd — APPROVED** [review]
Thanks a lot @gautamprikshit1 , great (and fast!) work!

**Pull request merged**